Cython/Compiler/Parsing.py

```diff
@@ -1161,6 +1168,16 @@ def p_f_string_expr(s, unicode_value, pos, starting_index, is_raw):
         error(_f_string_error_pos(pos, unicode_value, starting_index),
               "empty expression not allowed in f-string")
 
+    if terminal_char == '=':
+        i += 1
+        while unicode_value[i].isspace():
+            if i >= size:
+                break  # error will be reported below
+            i += 1
```

Comment:
This looks like `i` might exceed the length of `unicode_value` here if the string ends with a `=` or spaces. This could trigger a compiler crash rather than an error report, either right in the `while` condition or further down when reading the new `terminal_char`.

Reply:
Ending with an `=` or ending with spaces (or a combination of both) works perfectly fine. I did notice however that a malformed string does cause a compiler crash. Will fix

Code:
```
a = "test"
b = f"{a="
```

Reply:
Fixed